sra_experiment/nbconverted/DEG_validation.py

```diff
@@ -0,0 +1,311 @@
+
+# In[10]:
+
+
+expression_data = pd.read_csv(paths.PAO1_GE_DE, sep="\t", index_col=0)
+expression_data.head()
+
+
+# In[11]:
+
+
+# Group samples as PAO1 or PA14 based on experiment metadata
+sample_annot_file = paths.SAMPLE_ANNOT_TEST
+pao1_ids, pa14_ids = utils.get_sample_grps(sample_annot_file)
+
+
+# In[12]:
+
+
+# Split expression by genotype
+pao1_expression = expression_data.loc[pao1_ids]
+pa14_expression = expression_data.loc[pa14_ids]
+
+
+# In[13]:
+
+
+# Get within sample correlation
+pao1_corr = pao1_expression.T.corr()
+pa14_corr = pa14_expression.T.corr()
+
+
+# In[14]:
+
+
+ax = sns.heatmap(
+    pao1_corr, 
+    vmin=-1, vmax=1, center=0,
+    cmap=sns.diverging_palette(20, 220, n=200),
+    square=True
+)
+ax.set_title("PAO1 sample correlation")
+
+
+# In[15]:
+
+
+ax = sns.heatmap(
+    pa14_corr, 
+    vmin=-1, vmax=1, center=0,
+    cmap=sns.diverging_palette(20, 220, n=200),
+    square=True
+)
+ax.set_title("PA14 sample correlation")
+
+
+# In[16]:
+
+
+# Get mean expression for pao1 and pa14
+mean_pao1 = pao1_expression.mean()
+mean_pa14 = pa14_expression.mean()
+
+pao1_v_pa14_df = pd.DataFrame(data={'pao1_mean': mean_pao1.values,
+                                   'pa14_mean': mean_pa14.values},
+                             index=pao1_expression.columns)
+
+pao1_v_pa14_df.head()
+
+
+# In[17]:
+
+
+sns.scatterplot(data=pao1_v_pa14_df, x='pao1_mean', y='pa14_mean')
+plt.plot([0,140000],[0,140000])
+
+
+# In[18]:
+
+
+# Rough calculation of the number of genes that differ
+# This is roughly what we would expect to get from the DE analysis
+
+std_x_eq_y = np.std(abs(pao1_v_pa14_df['pao1_mean']-pao1_v_pa14_df['pa14_mean']))
+gene_differences = pao1_v_pa14_df[abs(pao1_v_pa14_df['pao1_mean']-pao1_v_pa14_df['pa14_mean']) > std_x_eq_y]
+print(gene_differences.shape)
+genes_found_from_GE = list(gene_differences.index)
+gene_differences.head()
+
+
+# **Observations:**
+# * Looks like there is consistent gene expression patterns within sample-type (i.e. Both PAO1 samples have a similar gene expression profile, similarly for both PA14 samples) as expected
+# * Comparing the mean expression of PAO1 and PA14 we see that there are ~200 genes changed. This gives us some indication about what to expect for our DESeq analysis. However, we shouldn't expect the numbers to align because we are using different methods -- above we are comparing the raw gene expression values and looking for a threshold difference; below we DESeq fits the negative binomial model to the data and performs hypothesis testing to determine if there is a difference between the groups of samples.  
+
+# ### Differential expression analysis
+
+# In[19]:
+
+
+get_ipython().run_cell_magic('R', '', '# Select 59\n# Run one time\n#if (!requireNamespace("BiocManager", quietly = TRUE))\n#    install.packages("BiocManager")\n#BiocManager::install("DESeq2")')
+
+
+# In[20]:
+
+
+get_ipython().run_cell_magic('R', '', '# Load the DESeq2 library\nlibrary("DESeq2")')
```

Comment:
If you want, you can wrap the `library` call in a `suppressPackageStartupMessages` call to get rid of these warnings that happen when you load an R package. I don't think these warnings are particularly helpful, so probably fine to just suppress them.

Reply:
I had no idea about this! So nice